api/orchestrator/orchestrator.proto

```diff
@@ -434,34 +526,119 @@ message CloudService {
 
   string description = 3;
 
-  message Requirements { repeated string requirement_ids = 1; }
-
-  Requirements requirements = 4;
-
-  repeated MetricConfiguration metric_configurations = 5
+  repeated MetricConfiguration metric_configurations = 4
       [ (tagger.tags) = "gorm:\"many2many:cloud_service_metric_configurations;"
-                        "References:service_id\"" ];
+                        "foreignKey:id;"
+                        "references:service_id,metric_id\"" ];
 }
 
-// Requirement represents a certain requirement that needs to be fulfilled. It
-// could be a control in a certification catalog.
-message Requirement {
+message Catalog {
   string id = 1;
 
   string name = 2;
 
   string description = 3;
 
+  repeated Category categories = 4
+      [ (tagger.tags) = "gorm:\"constraint:OnDelete:CASCADE\"" ];
+}
+
+message Category {
+  string name = 1 [ (tagger.tags) = "gorm:\"primaryKey\"" ];
+
+  // Reference to the catalog this category belongs to.
+  string catalog_id = 2 [ (tagger.tags) = "gorm:\"primaryKey\"" ];
+
+  string description = 3;
+
+  repeated Control controls = 4 [
+    (tagger.tags) =
+        "gorm:\"foreignKey:category_name,category_"
+        "catalog_id;references:name,catalog_id;constraint:OnDelete:CASCADE\""
+  ];
+}
+
+// Control represents a certain Control that needs to be fulfilled. It could be
+// a Control in a certification catalog. It follows the OSCAL model. A
+// requirement in the EUCS terminology, e.g., is represented as the lowest
+// sub-control.
+message Control {
+  // A short name, e.g. OPS-01
+  // TODO(oxisto): name this "id"? or is it too confusing that this id is not
+  // unique? however, it is usually referred to as a control ID
+  string short_name = 1 [ (tagger.tags) = "gorm:\"primaryKey\"" ];
```

Comment:
Should we just name this "ID" as its commonly referred to in standards and have a comment saying that ID itself is not unique but just together with the other primary keys?

Reply:
Ok, let's use "id", that way we are also closer to the OSCAL terminology